src/languages/c.js

```diff
@@ -57,13 +57,14 @@ export default function(hljs) {
   const NUMBERS = {
     className: 'number',
     variants: [
-      { begin: '\\b(0b[01\']+)' },
-      { begin: '(-?)\\b([\\d\']+(\\.[\\d\']*)?|\\.[\\d\']+)((ll|LL|l|L)(u|U)?|(u|U)(ll|LL|l|L)?|f|F|b|B)' },
-      { begin: '(-?)(\\b0[xX][a-fA-F0-9\']+|(\\b[\\d\']+(\\.[\\d\']*)?|\\.[\\d\']+)([eE][-+]?[\\d\']+)?)' }
-    ],
+      { match: /\b(0b[01']+)/ },  
+      { match: /(-?)\b([\d']+(\.[\d']*)?|\.[\d']+)((ll|LL|l|L)(u|U)?|(u|U)(ll|LL|l|L)?|f|F|b|B)/ },  
+      { match: /(-?)(0[xX][a-fA-F0-9]+(?:'[a-fA-F0-9]+)*(?:\.[a-fA-F0-9]*(?:'[a-fA-F0-9]*)*)?(?:[pP][-+]?[0-9]+)?(l|L)?(u|U)?)/ },  
```

Comment:
Do we want need `\b` here also?

Reply:
i did see a redundant \b and ive removed it

Reply:
I thought these were needed to prevent false positives... like say `g2324` where the `234` would be identified as a number without the `\b`, no?

Reply:
![test](https://github.com/user-attachments/assets/254d04a6-3f41-470b-b3bb-1d302a95342f)
the current behavior seems to work fine but ig we can do with a \b just to be sure but wouldnt that take a performance hit. should i add \b just to be sure or is it fine ?

Reply:
<img width="84" alt="Screen Shot 2024-08-24 at 8 39 40 PM" src="https://github.com/user-attachments/assets/bfafef29-8998-49e2-bd3c-6913cce693f7">

An example of false positives, wouldn't these be valid variable names?  The `\b` is needed to prevent this.

> wouldnt that take a performance hit.

Not really, no - I don't think so - but we don't really talk about regex performance here outside of security concerns (run-away exponential regex, etc).

Reply:
> should i add \b

Yep. I think we need it.

Reply:
![image](https://github.com/user-attachments/assets/3ae5c9ac-21b3-4572-ab3d-d1efe0a2be9c)
@joshgoebel  ive fixed the false positive issue please review changes